### ISSUE TITLE
Overflow exception related to IptcProfile

### ISSUE THREAD

**Issue by eelke-at-bolt** [SixLabors/ImageSharp#1942]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

Loading the attached image causes en OverflowException
```
System.OverflowException: Arithmetic operation resulted in an overflow.
   at SixLabors.ImageSharp.Metadata.Profiles.Iptc.IptcProfile.Initialize()
   at SixLabors.ImageSharp.Metadata.Profiles.Iptc.IptcProfile..ctor(Byte[] data)
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoderCore.InitIptcProfile()
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoderCore.Identify(BufferedReadStream stream, CancellationToken cancellationToken)
   at SixLabors.ImageSharp.Formats.ImageDecoderUtilities.Identify(IImageDecoderInternals decoder, Configuration configuration, Stream stream)
   at SixLabors.ImageSharp.Formats.Jpeg.JpegDecoder.Identify(Configuration configuration, Stream stream)
   at SixLabors.ImageSharp.Image.InternalIdentity(Stream stream, Configuration config)
   at SixLabors.ImageSharp.Image.<>c__DisplayClass111_0.<Identify>b__0(Stream s)
   at SixLabors.ImageSharp.Image.WithSeekableStream[T](Configuration configuration, Stream stream, Func`2 action)
   at SixLabors.ImageSharp.Image.Identify(Configuration configuration, Stream stream, IImageFormat& format)
   at SixLabors.ImageSharp.Image.Identify(Configuration configuration, String filePath, IImageFormat& format)
   at SixLabors.ImageSharp.Image.Identify(String filePath, IImageFormat& format)
   at SixLabors.ImageSharp.Image.Identify(String filePath)
   at imageTest.Program.Main(String[] args) in C:\Users\nobody\source\repos\imageTest\Program.cs:line 14
```

### Steps to Reproduce

Load the attached file with `SixLabors.ImageSharp.Image.Load(filename);`

### System Configuration

- ImageSharp version: 1.0.4
- Other ImageSharp packages and versions: none
- Environment (Operating system, version and so on): Windows 10 (but also in linux based docker container)
- .NET Framework version: net6 on windows 10 (the linux container is on core 3.1)
- Additional information:

![005f02803c9843ec941ccc3bef68df6f](https://user-images.githubusercontent.com/8037266/149316023-48db223e-c77f-4a4a-a2dc-f4287f3b6b99.jpg)

**Comment by ynse01** [comment]
This image contains at least one Extended DataSet Tag. As shown on page 14 of the [IIMV4.2 specification](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Metadata/Profiles/IPTC/IIMV4.2_IPTC.pdf), the most significant bit of byte 4 of the header is used as a flag to indicate this.

However, in the class `IptcProfile` on line 284, this bit is interpreted as the signed bit for the data field length. This NOT the correct interpretation and the root cause for the failure to load this image. I was able to read this image correctly using Irfanview and Magick.

In other words, to load this image correctly, ImageSharp needs to support IPCT's Extended DataSet feature.

**Comment by brianpopow** [comment]
@ynse01 thank you for looking into this!

**Comment by brianpopow** [comment]
The provided testimage seems to have a invalid IPTC profile. The second IPTC value does not start with the expected `0x1c` tag marker, its 0 instead.

**Comment by brianpopow** [comment]
I could not find any example images which contain the extended data tag values, even on the [IPTC website](https://iptc.org/standards/photo-metadata/iptc-standard/) the provided example images does not contain such data. If someone has such images, that would be really helpful for implementing IPCT's Extended DataSet feature.

Otherwise, i have a fix for this issue here: https://github.com/SixLabors/ImageSharp/tree/bp/issue1942

**Comment by JimBobSquarePants** [comment]
@brianpopow I had a look at your fix. I'm happy with that for now. If IPTC can't supply an extended profile example and that example contains error then I doubt many are in circulation.

**Comment by brianpopow** [comment]
> @brianpopow I had a look at your fix. I'm happy with that for now. If IPTC can't supply an extended profile example and that example contains error then I doubt many are in circulation.

I really doubt Extended DataSet feature is really used anywhere. The spec says the octets 4 and 5 would specify the length of the data field, which would mean the data field octet count could be a variable sized int anything up to a max of `0x7FFF`, which is ridiculous. We should not try to support that. This seems to me to obscure.

**Comment by JimBobSquarePants** [comment]
Agreed
